test/integration/patients/worklist/worklist.js

```diff
@@ -2703,6 +2703,216 @@ context('worklist page', function() {
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .find('.js-select')
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v2'));
+
+        expect(storage.lastSelectedIndex).to.equal(null);
+      });
+
+    cy
+      .get('.worklist-list__filter-region')
+      .find('.js-cancel')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .find('.js-select')
+      .click();
+
+    cy
+      .get('[data-select-all-region]')
+      .find('.fa-square-minus')
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v2'));
+
+        expect(storage.lastSelectedIndex).to.equal(null);
+      });
+
+    cy
+      .get('.worklist-list__filter-region')
+      .find('.js-cancel')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .find('.js-select')
+      .click();
+
+    cy
+      .get('.worklist-list__filter-region')
+      .find('.js-cancel')
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v2'));
+
+        expect(storage.lastSelectedIndex).to.equal(null);
+      });
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .find('.js-select')
+      .click();
+
+    cy
+      .get('.list-page__header')
+      .find('[data-search-region] .js-input:not([disabled])')
+      .as('listSearch')
+      .focus()
+      .type('abcd')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v2'));
+
+        expect(storage.lastSelectedIndex).to.equal(null);
+      });
+
+    cy
+      .get('@listSearch')
+      .next()
+      .click();
+
+    cy
+      .get('.worklist-list__filter-region')
+      .find('.js-cancel')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .find('.js-select')
+      .click();
+
+    cy
+      .get('.worklist-list__toggle')
+      .contains('Actions')
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v2'));
+
+        expect(storage.lastSelectedIndex).to.equal(null);
+      });
+
+    cy
+      .get('.worklist-list__toggle')
+      .contains('Flows')
+      .click();
+
+    cy
+      .get('.worklist-list__filter-region')
+      .find('.js-cancel')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .find('.js-select')
+      .click();
+
+    cy
+      .navigate('/schedule');
+
+    cy
+      .go('back')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v2'));
+
+        expect(storage.lastSelectedIndex).to.equal(null);
+      });
+  });
+
```

Comment:
**Overview of these tests**:

1. Load the `/worklist/owned-by` page with the `Flows` version of the list shown and display 3 flows in the list.

2. Test the `click`+`shift` functionality in the list.
    - Multi-select from top to bottom in the list.
    - Multi-select from bottom to top in the list.
    - Verify that the `state.lastSelectedIndex` value is set properly when list items are selected (by examining `localStorage` value).
    
3. Verify that the `state.lastSelectedIndex` value is being properly cleared (set to `null`) in the correct situations.
    - When a list item is un-selected.
    - When the select all option is clicked.
    - When the bulk select cancel option is clicked.
    - When the list is filtered using the `Find in list…` input.
    - When the worklist page initially loads.
      - This is tested by selecting a list item, navigating to a new page, and then going back to the worklist where the `lastSelectedIndex` should equal `null` in `localStorage`.

**Questions I have**:

1. Is this a good place to put these tests in the codebase? Or should they go in the `test/integration/patients/worklist/bulk-edit.js` file?
2. I'm only testing the `Flows` version of the worklist in these tests. The `Flows` and `Actions` lists are almost identical, so I figured that would suffice?